sources/web/datalab/polymer/modules/Utils.ts

```diff
@@ -112,4 +112,20 @@ class Utils {
     }
   }
 
+  /**
+   * Called when an API call fails, so that a connection error message can be shown.
+   */
+  static connectionFailed() {
+    const host = <DatalabAppElement>document.querySelector('datalab-app');
+    host.showNotification('Failed to connect to the server.', true /*sticky*/);
+  }
+
+  /**
+   * Called when an API call succeeds, so that the connection error message can be hidden.
+   */
+  static connectionSucceeded() {
+    const host = <DatalabAppElement>document.querySelector('datalab-app');
+    host.hideNotification();
```

Comment:
Once you start using the toast for things other than connection failures, are you pretty confident that you will never be displaying some other toast when a connection succeeds and clears the toast? Maybe there are two XHRs outstanding after a connection failure, and one succeeds but then has some other error that shows a toast, but then the other completes just after that, clearing the non-connection-failure toast.